website/source/docs/internals/providers/new-data-source.html.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: "docs"
+page_title: "Creating Data Sources"
+sidebar_current: "docs-internals-provider-guide-new-data-source"
+description: |-
+  How to get started adding a data source to an existing provider.
+---
+
+# Creating Data Sources
+
+Data sources are components of providers that pull information from outside of
+Terraform's managed resources, to make it available in configuration.  They
+should exist in the same package as the provider.
+
+The built-in plugins for Terraform that are shipped as part of the codebase
+have a file naming guideline: they use
+`data_source_{api}_{data_source_name}.go` as the template to name files, and
```

Comment:
We use the convention `<THING>` in these cases. So `data_source_<API>_<DATA_SOURCE_NAME>` instead